### PR TITLE
Allow variable top relation label in a dependency graph.

### PR DESCRIPTION
**Pull request by dimazest**
This is  a possible (partial) fix for #944

### CODE REVIEW

File: nltk/test/dependency.doctest [hunk 1]
```diff
@@ -181,3 +181,36 @@ Non-Projective Dependency Parsing
...
+>>> print(dg.to_conll(10))  # doctest: +NORMALIZE_WHITESPACE
+1   I       _       NN      NN      _       2       nn      _       _
+2   shot    _       NN      NN      _       0       null    _       _
+3   an      _       AT      AT      _       2       dep     _       _
+4   elephant        _       NN      NN      _       7       nn      _       _
+5   in      _       NN      NN      _       7       nn      _       _
+6   my      _       NN      NN      _       7       nn      _       _
+7   pajamas _       NNS     NNS     _       3       dobj    _       _
+
+.. >>> import pytest; pytest.set_trace()
```
Comment (dimazest) [inline nltk/test/dependency.doctest]: has to be removed, i was in a hurry comiting changes

File: nltk/test/dependency.doctest [hunk 2]
```diff
@@ -181,3 +181,36 @@ Non-Projective Dependency Parsing
...
+
+>>> dg_str = """1       I       _       NN      NN      _       2       nn      _       _
+... 2   shot    _       NN      NN      _       0       null    _       _
+... 3   an      _       AT      AT      _       2       dep     _       _
+... 4   elephant        _       NN      NN      _       7       nn      _       _
+... 5   in      _       NN      NN      _       7       nn      _       _
+... 6   my      _       NN      NN      _       7       nn      _       _
+... 7   pajamas _       NNS     NNS     _       3       dobj    _       _
+... """
+>>> dg = DependencyGraph(dg_str, top_relation_label='null')
```
Comment (dimazest) [inline nltk/test/dependency.doctest:199]: here we tell DG that the top relation is `null`!

**Comment by stevenbird** [comment]
@dimazest, would you please let me know if this is still current, now that I have merged #944?

**Comment by dimazest** [comment]
It's outdated, as there is a hack implemented in `nltk.parse.malt.MaltParser.pretrained_model_sanity_checks()`.

What is even more important than having a flexible root node name is to support several roots https://github.com/nltk/nltk/pull/944#issuecomment-123619591

As a consequence, tree visualization code has to be changed. I suggest to always print the artificial root node (the one that  we currently pretend doesn't exist).

Other `DependencyCode` that uses `.root` needs adaptation as well (for example  `.triples()` https://github.com/dimazest/nltk/blob/top_relation/nltk/parse/dependencygraph.py#L354), not to mention code outside of the dependencygraph.py.

I can work towards support of several heads in this branch.

By the way a `parse.util` tests fails for me:

```
============================= test session starts ==============================
platform darwin -- Python 3.4.3 -- py-1.4.30 -- pytest-2.7.2
rootdir: /Users/dimazest/Documents/qmul/tools/src/nltk, inifile: pytest.ini
plugins: bdd, cov, xdist
collected 9 items

nltk/parse/util.py F

=================================== FAILURES ===================================
________________ [doctest] nltk.parse.util.taggedsent_to_conll _________________
080     """
081     A module to convert the a POS tagged document stream 
082     (i.e. list of list of tuples) and yield lines in CONLL format. 
083     This module yields one line per word and two newlines for end of sentence. 
084 
085     >>> from nltk import word_tokenize, sent_tokenize
086     >>> text = "This is a foobar sentence. Is that right?"
087     >>> sentences = [word_tokenize(sent) for sent in sent_tokenize(text)]
088     >>> for line in taggedsent_to_conll(sentences):
UNEXPECTED EXCEPTION: ValueError('too many values to unpack (expected 2)',)
Traceback (most recent call last):

  File "/Users/dimazest/miniconda3/lib/python3.4/doctest.py", line 1318, in __run
    compileflags, 1), test.globs)

  File "<doctest nltk.parse.util.taggedsent_to_conll[3]>", line 1, in <module>

  File "/Users/dimazest/Documents/qmul/tools/src/nltk/nltk/parse/util.py", line 103, in taggedsent_to_conll
    for (i, (word, tag)) in enumerate(sentence, start=1):

ValueError: too many values to unpack (expected 2)

/Users/dimazest/Documents/qmul/tools/src/nltk/nltk/parse/util.py:88: UnexpectedException
================== 8 tests deselected by '-knltk.parse.util' ===================
==================== 1 failed, 8 deselected in 0.91 seconds ====================
```

In my case `sentences` is just a list of strings: `['This', 'is', 'a', 'foobar', 'sentence', '.']`.

**Comment by alvations** [comment]
@dimazest, The input for  `nltk.parse.util.taggedsent_to_conll()` is a list of tagged sentence. Sorry that the docstring had some errors. Fixed it in https://github.com/alvations/nltk/commit/10da3b8f107f2b90ab047ddb9e58d5744306e7bb

**Comment by dimazest** [comment]
@alvations great!

**Comment by dimazest** [comment]
#1114 might benefit from this PR, should we consider merging it in?

**Comment by stevenbird** [comment]
I agree @dimazest. It doesn't break anything, just gives more generality.

**Comment by stevenbird** [comment]
The merge conflicts are too tricky for me to deal with. @dimazest would you please update this PR for me?

**Comment by dimazest** [comment]
@stevenbird sure I'll take care of the merge conflict, just wanted to get a green light!

**Comment by dimazest** [comment]
Done!

**Comment by stevenbird** [comment]
Thanks @dimazest, but I can't check 114 files. I'll cherry-pick the changes to malt.py

**Comment by stevenbird** [comment]
Thanks @dimazest

**Pull request closed without merging**